### PR TITLE
Do not process URLs that use file or ftp scheme

### PR DESCRIPTION
**Pull request by koops76**
See https://developer.chrome.com/extensions/match_patterns.

### CODE REVIEW

**Comment by J0WI** [comment]
`*://*/*` does still match `ftp://foo/bar`. Maybe `http` is enough? (see #10843)
cc @Hainish

**Comment by koops76** [comment]
@J0WI No: https://developer.chrome.com/extensions/match_patterns.

**Comment by koops76** [comment]
@J0WIc `http` isn't currently enough.

File: chromium/background.js [hunk 1]
```diff
@@ -573,7 +573,7 @@ function onBeforeRedirect(details) {
 
 // Registers the handler for requests
 // See: https://github.com/EFForg/https-everywhere/issues/10039
-wr.onBeforeRequest.addListener(onBeforeRequest, {urls: ["<all_urls>"]}, ["blocking"]);
+wr.onBeforeRequest.addListener(onBeforeRequest, {urls: ["*://*/*"]}, ["blocking"]);
```
Comment (Hainish) [inline chromium/background.js]: This was changed to `<all_urls>` due to reasoning I describe in https://github.com/EFForg/https-everywhere/issues/10039#issuecomment-307276140.  If this changes, we'll have to test this in `webextensions` as well.

Comment (koops76) [inline chromium/background.js]: So, are we going to wait until `webextensions` becomes stable?

Comment (Hainish) [inline chromium/background.js]: I just tested it in the `webextensions` branch, looks okay.

Comment (koops76) [inline chromium/background.js]: Okay, merging?

Comment (koops76) [inline chromium/background.js]: Also try `["http://*/*", "https://*/*"]` if it doesn't work.

File: chromium/manifest.json [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
-        "storage", 
-        "<all_urls>"
-    ], 
+        "webNavigation",
+        "webRequest",
+        "webRequestBlocking",
+        "tabs",
+        "cookies",
+        "storage",
+        "*://*/*"
```
Comment (Hainish) [inline chromium/manifest.json]: Only change this line in the file, not all the other lines that remove whitespace.  These will just be added back with [`utils/setversion.py`](https://github.com/EFForg/https-everywhere/blob/master/utils/setversion.py) anyway.  If this bothers you, open a new PR that removes the whitespaces via that script.

Comment (koops76) [inline chromium/manifest.json]: @Hainish Done.

Comment (Hainish) [inline chromium/manifest.json]: I just fixed the underlying problem of whitespace generation in 92980455d5a6869803ac88f941bce0f0a33f7c8c

Comment (koops76) [inline chromium/manifest.json]: @Hainish Would a rewrite of Python parts of this project in Node and rewrite of `hsts-prune` that will run on Windows and have better code quality and less dependencies be welcome? To be honest currently `hsts-prune` is a mess.

Comment (koops76) [inline chromium/manifest.json]: @Hainish Also see https://github.com/koops76/https-everywhere-rules and #11107. Webpack will allow us to extract ruleset parsing and applying into easily testable modules that can be tested using Mocha and Node independently of browser. I already developed a library that can be shared by both ruleset testing scripts and the extension. It is in the need of improvement though. You can also try improving it.

Comment (Hainish) [inline chromium/manifest.json]: @koops76 to be honest I don't see the need to rewrite `hsts-prune`.  We already have an open ticket for rewriting the tools in `node`: https://github.com/EFForg/https-everywhere/issues/9993

Comment (Hainish) [inline chromium/manifest.json]: A rewrite of the tools will require a lot of my attention to review, which I can't provide until after `webextensions` is deployed next month.

Comment (koops76) [inline chromium/manifest.json]: @Hainish It freezes under Windows.

Comment (koops76) [inline chromium/manifest.json]: @Hainish OK.

**Comment by Hainish** [comment]
Merged in https://github.com/EFForg/https-everywhere/commit/8e302ec41139808bab2ddb63a2dca11d1b78c168

**Pull request closed without merging**